sentry_sdk/integrations/celery/__init__.py

```diff
@@ -311,20 +311,38 @@ def _inner(*args, **kwargs):
     return _inner  # type: ignore
 
 
+def _set_messaging_destination_name(task, span):
+    # type: (Any, Span) -> None
+    """Set "messaging.destination.name" tag for span"""
+    with capture_internal_exceptions():
+        delivery_info = task.request.delivery_info
+        routing_key = delivery_info.get("routing_key")
+        if delivery_info.get("exchange") == "" and routing_key is not None:
+            # Empty exchange indicates the default exchange, meaning the tasks
+            # are sent to the queue with the same name as the routing key.
+            span.set_data("messaging.destination.name", routing_key)
+
+
 def _wrap_task_call(task, f):
     # type: (Any, F) -> F
 
     # Need to wrap task call because the exception is caught before we get to
     # see it. Also celery's reported stacktrace is untrustworthy.
 
     # functools.wraps is important here because celery-once looks at this
-    # method's name.
+    # method's name. @ensure_integration_enabled internally calls functools.wraps,
+    # but if we ever remove the @ensure_integration_enabled decorator, we need
+    # to add @functools.wraps(f) here.
     # https://github.com/getsentry/sentry-python/issues/421
-    @wraps(f)
+    @ensure_integration_enabled(CeleryIntegration, f)
     def _inner(*args, **kwargs):
         # type: (*Any, **Any) -> Any
         try:
-            return f(*args, **kwargs)
+            with sentry_sdk.start_span(
+                op=OP.QUEUE_TASK_CELERY, description=task.name
```

Comment:
Is there a better `op` we could use here? `queue.task.celery` seemed like the most fitting `op`; however, since the parent transaction also has `op` set to `queue.task.celery`, this leads to the span having the same `op` as its parent, which could be somewhat confusing

Reply:
Will be `queue.process` if I get the context here right.
We'll put `celery` on `messaging.system`.

Reply:
So this would be a new span `op` @cleptric? I do not see `queue.process` listed anywhere on our [list of span `op`s](https://develop.sentry.dev/sdk/performance/span-operations/).